doc/whatsnew/fragments/10334.bugfix

```diff
@@ -0,0 +1,3 @@
+Using a slice as a class decorator now raise a 'not-callable' message instead of crashing pylint.
```

Comment:
```suggestion
Using a slice as a class decorator now raise a 'not-callable' message instead of crashing pylint. A lot of checks that dealt with decorators (too many to list) are now shortcut if the decorator can't immediately be inferred to a function or class definition.
```
There's really a lot of use for ``decorated_with`` in a loooot of checks. How about we use hasattr in the backported fix and only do the optim in 4.0 ?